genie-web/src/main/java/com/netflix/genie/web/agent/launchers/impl/TitusAgentLauncher.java

```diff
@@ -0,0 +1,226 @@
+
+    private final RestTemplate restTemplate;
+    private final Cache<String, String> healthIndicatorCache;
+    private final GenieHostInfo genieHostInfo;
+    private final TitusAgentLauncherProperties titusAgentLauncherProperties;
+    private final MeterRegistry registry;
+
+    /**
+     * Constructor.
+     *
+     * @param restTemplate                 the rest template
+     * @param healthIndicatorCache         a cache to store metadata about recently launched jobs
+     * @param genieHostInfo                the metadata about the local server and host
+     * @param titusAgentLauncherProperties the configuration properties
+     * @param registry                     the metric registry
+     */
+    public TitusAgentLauncher(
+        final RestTemplate restTemplate,
+        final Cache<String, String> healthIndicatorCache,
+        final GenieHostInfo genieHostInfo,
+        final TitusAgentLauncherProperties titusAgentLauncherProperties,
+        final MeterRegistry registry
+    ) {
+        this.restTemplate = restTemplate;
+        this.healthIndicatorCache = healthIndicatorCache;
+        this.genieHostInfo = genieHostInfo;
+        this.titusAgentLauncherProperties = titusAgentLauncherProperties;
+        this.registry = registry;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Optional<JsonNode> launchAgent(
+        final ResolvedJob resolvedJob,
+        @Nullable final JsonNode requestedLauncherExt
+    ) throws AgentLaunchException {
+
+        final String jobId = resolvedJob.getJobSpecification().getJob().getId();
+
+        final TitusBatchJobRequest titusJobRequest = this.createJobRequest(resolvedJob);
+        final Set<Tag> tags = Sets.newHashSet();
+
+        String titusJobId = "unknown";
+
+        final long start = System.nanoTime();
+        try {
+            final TitusBatchJobResponse titusResponse = this.restTemplate.postForObject(
+                this.titusAgentLauncherProperties.getEndpoint(),
+                titusJobRequest,
+                TitusBatchJobResponse.class
+            );
+
+            if (titusResponse == null) {
+                throw new AgentLaunchException("Failed to request creation of Titus job for job " + jobId);
+
+            } else if (StringUtils.isBlank(titusResponse.getId())) {
+                throw new AgentLaunchException(
+                    "Failed to create titus job for job " + jobId
+                        + " - " + titusResponse.getStatusCode() + ": "
+                        + titusResponse.getMessage()
+                );
+
+            }
+
+            titusJobId = titusResponse.getId();
+
+            log.info("Created Titus job {} to execute job {}", titusJobId, jobId);
+
+            MetricsUtils.addSuccessTags(tags);
+
+            final JsonNode launcherExt = JsonNodeFactory.instance.objectNode()
+                .put("launcher_class", this.getClass().getCanonicalName())
```

Comment:
`this.getClass().getCononnicalName()` could be a static constant as could the various keys used here